bigbluebutton-web/src/java/org/bigbluebutton/api/RecordingService.java

```diff
@@ -1,259 +1,405 @@
+            for (int f = 0; f < recordings.size(); f++) {
+                if (!ids.contains(recordings.get(f).getName()))
+                    ids.add(recordings.get(f).getName());
+            }
+        }
+
+        return ids;
+    }
+
+    private Set<String> getAllRecordingIds(List<File> recs) {
+        Set<String> ids = new HashSet<String>();
+
+        Iterator<File> iterator = recs.iterator();
+        while (iterator.hasNext()) {
+            ids.add(iterator.next().getName());
+        }
+
+        return ids;
+    }
+
+    private List<Recording> getRecordingsForPath(String id, List<File> recordings) {
+        List<Recording> recs = new ArrayList<Recording>();
+
+        Iterator<File> iterator = recordings.iterator();
+        while (iterator.hasNext()) {
+            File recording = iterator.next();
+            if (recording.getName().startsWith(id)) {
+                Recording r = getRecordingInfo(recording);
+                if (r != null)
+                    recs.add(r);
+            }
+        }
+        return recs;
+    }
+
+    private Recording getRecordingInfo(File dir) {
+        Recording rec = recordingServiceHelper.getRecordingInfo(dir);
+        return rec;
+    }
+
+    private void deleteRecording(String id, String path) {
+        String[] format = getPlaybackFormats(path);
+        for (int i = 0; i < format.length; i++) {
+            List<File> recordings = getDirectories(path + File.separatorChar + format[i]);
+            for (int f = 0; f < recordings.size(); f++) {
+                if (recordings.get(f).getName().equals(id)) {
+                    deleteDirectory(recordings.get(f));
+                    createDirectory(recordings.get(f));
+                }
+            }
+        }
+    }
+
+    private void createDirectory(File directory) {
+        if (!directory.exists())
+            directory.mkdir();
+    }
+
+    private void deleteDirectory(File directory) {
+        /**
+         * Go through each directory and check if it's not empty. We need to
+         * delete files inside a directory before a directory can be deleted.
+         **/
+        File[] files = directory.listFiles();
+        for (int i = 0; i < files.length; i++) {
+            if (files[i].isDirectory()) {
+                deleteDirectory(files[i]);
+            } else {
+                files[i].delete();
+            }
+        }
+        // Now that the directory is empty. Delete it.
+        directory.delete();
+    }
+
+    private List<File> getDirectories(String path) {
+        List<File> files = new ArrayList<File>();
+        try {
+            DirectoryStream<Path> stream = Files.newDirectoryStream(FileSystems.getDefault().getPath(path));
+            Iterator<Path> iter = stream.iterator();
+            while (iter.hasNext()) {
+                Path next = iter.next();
+                files.add(next.toFile());
+            }

```

Comment:
Related to this comment: https://github.com/bigbluebutton/bigbluebutton/pull/3001#issuecomment-181380690

Found this: https://docs.oracle.com/javase/7/docs/api/java/nio/file/DirectoryStream.html
>  A DirectoryStream is opened upon creation and is closed by invoking the close method. Closing a directory stream releases any resources associated with the stream. Failure to close the stream may result in a resource leak. The try-with-resources statement provides a useful construct to ensure that the stream is closed:
>
>   Path dir = ...
>   try (DirectoryStream<Path> stream = Files.newDirectoryStream(dir)) {
>       for (Path entry: stream) {
>           ...
>       }
>   }
So we should revisit this code portion to release the resources properly.